data_gateway/packet_reader.py

```diff
@@ -97,7 +99,21 @@ def read_packets(self, serial_port, stop_when_no_more_data=False):
 
         with self.uploader:
             with self.writer:
+                packet_queue = queue.Queue()
+                error_queue = queue.Queue()
+
+                parser_thread = threading.Thread(
+                    target=self._parse_payload,
+                    kwargs={"packet_queue": packet_queue, "error_queue": error_queue},
+                    daemon=True,
+                )
+
+                parser_thread.setName("ParserThread")
+                parser_thread.start()
+

```

Comment:
I thought you were aiming to create a new parser thread for each packet? Perhaps that's not possible...?

Reply:
This is the PR from before I suggested even more threading. My plan was to merge this and then work on dispatching more threads if possible